secrets/secrets.go

```diff
@@ -0,0 +1,79 @@
+package secrets
+
+import (
+	"errors"
+	"sync"
+)
+
+var (
+	// ErrNotImplemented default secrets in OSD
+	ErrNotImplemented = errors.New("Not Implemented")
+	// ErrKeyEmpty returned when the secrety key provided is empty
+	ErrKeyEmpty = errors.New("Secret key cannot be empty")
+	// ErrNotAuthenticated returned when not authenticated with secrets endpoint
+	ErrNotAuthenticated = errors.New("Not authenticated with the secrets endpoint")
+	// ErrInvalidSecretId returned when no secret data is found associated with the id
+	ErrInvalidSecretId = errors.New("No Secret Data found for Secret Id")
+	// ErrEmptySecretData returned when no secret data is provided to store the secret
+	ErrEmptySecretData = errors.New("Secret data cannot be empty")
+	// ErrSecretExists returned when a secret for the given secret id already exists
+	ErrSecretExists = errors.New("Secret ID already exists")
+)
+
+type Secrets interface {
+	// SecretLogin create session with secret store
+	SecretLogin(secretType string, secretConfig map[string]string) error
+	// SecretSetDefaultSecretKey  sets the cluster wide secret key
+	SecretSetDefaultSecretKey(secretKey string, override bool) error
+	// SecretGetDefaultSecretKey returns cluster wide secret key's value
```

Comment:
this does not return the key's value, but the cluster wide key itself

Reply:
updated comment